src/externalized/scripting/FunctionsLibrary.cc

```diff
@@ -70,6 +71,11 @@ void PlaceItem(const INT16 sGridNo, OBJECTTYPE* const pObject, const INT8 ubVisi
 	AddItemToPool(sGridNo, pObject, static_cast<Visibility>(ubVisibility), 0, 0, 0);
 }
 
+MERCPROFILESTRUCT* GetMercProfile(const UINT8 ubProfileID)
+{
+	return &(GetProfile(ubProfileID));
```

Comment:
Why is it returning the address, not the pointer itself?

Reply:
The C function `GetProfile` returns by reference. Here we convert the reference to pointer, because Sol/Lua can only pass by value. And we want Lua to have the pointer (instead of a copy) so scripts can modify the data.